test/test_array.cpp

```diff
@@ -0,0 +1,275 @@
+#include "ports-of-call/array.hpp"
+#include "ports-of-call/portability.hpp"
+
+#ifndef CATCH_CONFIG_FAST_COMPILE
+#define CATCH_CONFIG_FAST_COMPILE
+#include <catch2/catch_test_macros.hpp>
+#endif
+
+#include <iostream>
+
+using namespace PortsOfCall;
+
+TEST_CASE("array nominal element access (GPU)", "[array][GPU]") {
+  // Declare the array
+  constexpr int N = 16;
+  array<double, N> arr;
+
+  int count = 0;
+  auto func = PORTABLE_LAMBDA(const int i, int &count) mutable {
+    // Fill the array
+    arr[i] = i + 1;
```

Comment:
I think the fill has to be done on host but it can be checked on device, because lambda will do a const capture.